src/experiments.ts

```diff
@@ -52,36 +37,80 @@ emulator.printMempool()
 console.log("End of mempool to check")
 emulator.printUtxos(emulator.getUtxos())
 
-// Bunch of useful functions, should we have them in the Emulator class?
-function createRandomInitialUtxos(numUtxos: number): IUTxO[] {
-    const utxos: IUTxO[] = []
+
+/**
+ * Create a list of random UTxOs with dynamically calculated ADA values based on transaction size.
+*/
+function createRandomInitialUtxos(
+    numUtxos: number, 
+    maxTxSizeBytes: number = 16384,
+    debugLevel: number = 1
+): IUTxO[] {
+
+    const utxos: IUTxO[] = [];
+
     for (let i = 0; i < numUtxos; i++) {
-        // (typeof id === "string" && bytestring_1.ByteString.isValidHexValue(id) && (id.length === 64))
-        const utxoHash = "a".repeat(63) + i.toString()
-        // console.log(utxoHash)
-        const address = generateRandomBech32Address() // Have a way to generate random Bech32 addresses
-        const utxoInit: IUTxO = createInitialUTxO(100000000n, address, utxoHash);
-        utxos.push(utxoInit)
+        const utxoHash = generateRandomTxHash(i); // Unique hash per UTxO
+        const address = generateRandomBech32Address();
+        const value = 100000000n; // 100 ADA (example value)
```

Comment:
the "100 Ada" should be read from an argument, either with ranges and a random generation or just a fixed value

Reply:
Applied